### PR TITLE
Fixed dSYM upload with s3

### LINKED ISSUE #1461: dSYM isn't uploaded to s3
When adding a "dSYM" property for s3 action, it is not uploaded among the other files.

### PR DESCRIPTION
**Pull request by oronbz**
resolves #1461

### CODE REVIEW

File: lib/fastlane/actions/s3.rb [hunk 1]
```diff
@@ -254,6 +262,39 @@ def self.upload_plist_and_html_to_s3(s3_access_key, s3_secret_access_key, s3_buc
         Helper.log.info "Successfully uploaded ipa file to '#{Actions.lane_context[SharedValues::S3_IPA_OUTPUT_PATH]}'".green
       end
 
+      def self.upload_dsym(s3_access_key, s3_secret_access_key, s3_bucket, s3_region, dsym_file_name, dsym_file_data)
+        Actions.verify_gem!('aws-sdk')
+        require 'aws-sdk'
+        if s3_region
```
Comment (oronbz) [inline lib/fastlane/actions/s3.rb]: rubocop is cool with it

**Comment by oronbz** [comment]
The current action did nothing with the dsym parameter.
I've added a similar implementation to the other files.

File: lib/fastlane/actions/s3.rb [hunk 2]
```diff
@@ -254,6 +262,39 @@ def self.upload_plist_and_html_to_s3(s3_access_key, s3_secret_access_key, s3_buc
...
+
+        # When you enable versioning on a S3 bucket,
+        # writing to an object will create an object version
+        # instead of replacing the existing object.
+        # http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/S3/ObjectVersion.html
+        if dsym_obj.kind_of? AWS::S3::ObjectVersion
+          dsym_obj = dsym_obj.object
+        end
+
+        # Setting actionand environment variables
```
Comment (ohwutup) [inline lib/fastlane/actions/s3.rb]: space between `action` and `and`

**Comment by ohwutup** [comment]
Thanks for this @oronbz! Just one little typo above, and if you wouldn't mind squashing after you fix that and we will be good to go!

File: lib/fastlane/actions/s3.rb (hunk 2 above)
Comment (oronbz) [inline lib/fastlane/actions/s3.rb]: This comment was already there i just copied it to another method (:
Anyway, incoming update

Comment (ohwutup) [inline lib/fastlane/actions/s3.rb]: oh! my bad! no worries!

**Comment by oronbz** [comment]
Thanks @ohwutup, fixed and squashed.

**Comment by ohwutup** [comment]
WORD UP! Thanks!!

**Pull request merged**